book/src/interface_builder.md

```diff
@@ -0,0 +1,135 @@
+Again, composite templates are described by `xml` files.
+
+<span class="filename">Filename: listings/interface_builder/2/window/window.ui</span>
+```xml
+{{#rustdoc_include ../listings/interface_builder/2/window/window.ui}}
+```
+
+Even the content seems to be nearly the same.
+Before, we described a pre-existing widget.
+
+```xml
+<object class="GtkApplicationWindow" id="window">
+```
+
+Now, we create a custom widget and let it inherit from a pre-existing one.
+
+```xml
+<template class="MyGtkAppWindow" parent="GtkApplicationWindow">
+```
+
+Within our code we also have to create a custom widget inheriting from `gtk::ApplicationWindow` to make use of our template.
+
+
+<span class="filename">Filename: listings/interface_builder/2/window/mod.rs</span>
+```rust,no_run
+{{#rustdoc_include ../listings/interface_builder/2/window/mod.rs}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+In the private struct, we then add the derive macro `gtk::CompositeTemplate`.
+We also specify that the template information comes from a file `window.ui` in the same folder.
+Additionally, we can hold a reference to any child of which we specified the `id` in the template.
+Since we need the button later on, we add it to the struct.
+
+<span class="filename">Filename: listings/interface_builder/2/window/imp.rs</span>
+```rust,no_run
+{{#rustdoc_include ../listings/interface_builder/2/window/imp.rs:object}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+Within the `ObjectSubclass` trait, we make sure that `NAME` corresponds to `class` in the template and `ParentType` corresponds to `parent` in the template.
+We also bind and initialize the template in `class_init` and `instance_init`.
+
+<span class="filename">Filename: listings/interface_builder/2/window/imp.rs</span>
+```rust,no_run
+{{#rustdoc_include ../listings/interface_builder/2/window/imp.rs:subclass}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+Finally, we connect the callback to the "clicked" signal of `button` within `constructed`.
+The button is easily available thanks to the stored reference in `self`.
+
+<span class="filename">Filename: listings/interface_builder/2/window/imp.rs</span>
+```rust,no_run
+{{#rustdoc_include ../listings/interface_builder/2/window/imp.rs:object_impl}}
+# // Please ignore this line
+# // It is only there to make mdbook happy
+# fn main() {}
+```
+
+With composite templates, `main.rs` actually becomes more concise.
+
+<span class="filename">Filename: listings/interface_builder/2/main.rs</span>
+```rust,no_run
+{{#rustdoc_include ../listings/interface_builder/2/main.rs:build_ui}}
+```
+Also with regard to capabilities, we get the best of both worlds.
+
+Thanks to custom widgets we can
+- keep state and part of it as properties,
+- add signals as well as
+- override behavior.
+
+Thanks to composite templates we can
```

Comment:
It also allows splliting code into smaller widgets. It would be also nice to mention Buildable, which are the objects you can instantiate through a gtk::Builder and so in a composite template as well. the Buildable interface is what tells to gtk how to transform what you wrote in xml to an object or whatever you want.

Reply:
> It also allows splliting code into smaller widgets

Can you give an example? 


Regarding interface. You want me to add a sentence like "if your widget implements BuildableExt trait then you can use it with Builder"?